WordPress/Classes/ViewRelated/Aztec/New Group/AztecVerificationPromptHelper.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+
+internal class AztecVerificationPromptHelper: NSObject {
+
+    private let accountService: AccountService
+    private let wpComAccount: WPAccount
+
+    private weak var displayedAlert: FancyAlertViewController?
+    private var completionBlock: AztecVerificationPromptCompletion?
+
+    typealias AztecVerificationPromptCompletion = (Bool) -> ()
+
+    init?(account: WPAccount?) {
+        guard let passedAccount = account,
+              let managedObjectContext = account?.managedObjectContext else {
+                return nil
+        }
+
+        self.accountService = AccountService(managedObjectContext: managedObjectContext)
+
+        guard accountService.isDefaultWordPressComAccount(passedAccount),
+              !passedAccount.emailVerified.boolValue else {
+                // if the post the user is trying to compose isn't on a WP.com account,
+                // or they're already verified, then the verification prompt is irrelevant.
+                return nil
+        }
+
+        self.wpComAccount = passedAccount
+
+        super.init()
+
+        NotificationCenter.default.addObserver(forName: .UIApplicationDidBecomeActive, object: nil, queue: nil) { [weak self] _ in
+            self?.updateVerificationStatus()
+        }
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
```

Comment:
As mentioned above, when in use with the Blocks API, this line would actually do nothing (and the object is expected to remain registered).